lib/mix/lib/mix/tasks/help.ex

```diff
@@ -194,13 +205,35 @@ defmodule Mix.Tasks.Help do
   end
 
   defp build_alias_doc_list(aliases) do
-    Enum.reduce(aliases, {[], 0}, fn {alias_name, _task_name}, {docs, max} ->
-      doc = "Alias defined in mix.exs"
+    Enum.reduce(aliases, {[], 0}, fn {alias_name, task}, {docs, max} ->
+      doc = alias_doc(task)
       task = "mix " <> alias_name
       {[{task, doc} | docs], max(byte_size(task), max)}
     end)
   end
 
+  defp alias_doc(task) do
+    "Alias for " <> format_alias_doc(task)
+  end
+
+  defp format_alias_doc(task) when is_list(task),
+    do: Enum.map_join(task, ", ", &format_alias_task/1)
+
+  defp format_alias_doc(task), do: format_alias_doc([task])
+
+  defp format_alias_task(task) when is_binary(task), do: task
```

Comment:
```suggestion
  defp format_alias_task(task) when is_binary(task),
    do: hd(:binary.split(task, " "))
```

Reply:
I think this is better because some aliases can be quite long, no?

Reply:
`mix help` does not use `OptionParser`, I would need to refactor it in order to properly support a flag. I could do it in a separate PR if you agree.

Reply:
Having the short version by default may be confusing, for example:

```
aliases: [
     "test.db": [test --only db],
     "test.api": [test --only api],
     "test.integration": [test --only integration]

]
```

would all give the same help message (*Alias for test*)

Reply:
But I am worried that including the whole thing will be too long and it will most often get trimmed. I guess since they get trimmed, that's fine? We gotta start somewhere anyway, so let's start with the least code. :)